Formula/dafny.rb

```diff
@@ -21,10 +22,12 @@ class Dafny < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "25d87dc6b3a2041668a3e4100d3e44c61bee2a772a8405fdc8fc63d1ac35a200"
   end
 
-  depends_on "gradle" => :build
-  depends_on "python@3.10" => :build # for z3
-  depends_on "dotnet"
-  depends_on "openjdk@11"
+  depends_on "python@3.11" => :build # for z3
+  depends_on "dotnet@6"
+  # We use the latest compatible Java version that is compatible with dafny's gradlew
+  # https://github.com/dafny-lang/dafny/blob/v#{version}/Source/DafnyRuntime/DafnyRuntimeJava/gradle/wrapper/gradle-wrapper.properties
+  # https://docs.gradle.org/current/userguide/compatibility.html
+  depends_on "openjdk@17"
```

Comment:
As far as I could tell, build is running `./gradlew` as part of C# project build. Currently, this is Gradle 7.4.2 which is fully compatible with up to OpenJDK 17.